scrape/manager.go

```diff
@@ -99,53 +101,55 @@ func (m *Manager) ApplyConfig(cfg *config.Config) error {
 
 // TargetMap returns map of active and dropped targets and their corresponding scrape config job name.
 func (m *Manager) TargetMap() map[string][]*Target {
-	m.mtx.Lock()
-	defer m.mtx.Unlock()
+	m.RLock()
+	defer m.RUnlock()
 
 	targets := make(map[string][]*Target)
 	for jobName, sp := range m.scrapePools {
-		sp.mtx.RLock()
+		sp.Lock()
 		for _, t := range sp.targets {
 			targets[jobName] = append(targets[jobName], t)
 		}
 		targets[jobName] = append(targets[jobName], sp.droppedTargets...)
-		sp.mtx.RUnlock()
+		sp.Unlock()
 	}
 
 	return targets
 }
 
 // Targets returns the targets currently being scraped.
 func (m *Manager) Targets() []*Target {
-	m.mtx.Lock()
-	defer m.mtx.Unlock()
+	m.RLock()
+	defer m.RUnlock()
 
 	var targets []*Target
 	for _, p := range m.scrapePools {
-		p.mtx.RLock()
+		p.RLock()
 		for _, tt := range p.targets {
 			targets = append(targets, tt)
 		}
-		p.mtx.RUnlock()
+		p.RUnlock()
 	}
 
 	return targets
 }
 
 // DroppedTargets returns the targets dropped during relabelling.
 func (m *Manager) DroppedTargets() []*Target {
-	m.mtx.Lock()
-	defer m.mtx.Unlock()
+	m.RLock()
+	defer m.RUnlock()
 	var droppedTargets []*Target
 	for _, p := range m.scrapePools {
-		p.mtx.RLock()
+		p.RLock()
 		droppedTargets = append(droppedTargets, p.droppedTargets...)
-		p.mtx.RUnlock()
+		p.RUnlock()
 	}
 	return droppedTargets
 }
 
 func (m *Manager) reload(t map[string][]*targetgroup.Group) {
+	m.Lock()
```

Comment:
This lock is too coarse, it'll hold up the UI while we're waiting for scrape loops to finish scraping.